File: src/main/java/com/metlife/santa/core/Vixen.java
package com.metlife.santa.core;

import com.metlife.santa.config.DasherConfig;
import org.codehaus.jackson.type.TypeReference;
import java.io.IOException;

public class Vixen extends ReindeerBase {

    public ReindeerBase init(String file) throws IOException {

        TypeReference<DasherConfig> typeRef
                = new TypeReference<DasherConfig>() {};

        init(typeRef,file);
        return this;
    }

    public void process(){
        this.rddInput.collect().forEach(o->{
            System.out.println(o);
        });
    }

    public void chain(ReindeerBase previous){
        this.rddInput=previous.rddOutput;
    }

}

File: src/main/scala/Vixen.scala
import java.util

import com.metlife.santa.config.VixenConfig
import org.apache.spark.rdd.RDD

class Vixen extends ReindeerBase{

  def init(file: String) = {
    initReindeer(classOf[VixenConfig],file)
  }

  def process(inputRDD:RDD[util.Map[String, AnyRef]]) = {
    inputRDD.collect.foreach(println)
  }
}
